engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobBatchCollector.java

```diff
@@ -86,10 +86,9 @@ Either<TooLargeJob, Integer> collectJobs(final TypedRecord<JobBatchRecord> recor
 
           // the expected length is based on the current record's length plus the length of the job
           // record we would add to the batch, the number of bytes taken by the additional job key,
-          // as well as one byte required per job key for its type header. if we ever add more, this
-          // should be updated accordingly.
+          // as well as an 8 KB buffer.
           final var jobRecordLength = jobRecord.getLength();
-          final var expectedEventLength = record.getLength() + jobRecordLength + Long.BYTES + 1;
+          final var expectedEventLength = record.getLength() + jobRecordLength + (1024 * 8);
```

Comment:
🔧  We can put `1024 * 8` in a variable to avoid replication in the tests. Or even better, we can move that into the `EngineConfiguration` file and use it where needed